src/test/java/spoon/test/model/IncrementalBuildTool.java

```diff
@@ -0,0 +1,126 @@
+package spoon.test.model;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.filefilter.NameFileFilter;
+import org.apache.commons.io.filefilter.TrueFileFilter;
+
+import spoon.reflect.declaration.CtPackage;
+import spoon.reflect.declaration.CtType;
+import spoon.reflect.factory.Factory;
+
+public class IncrementalBuildTool {
+
+	private File mOldBinaryOutputDirectory;
+	private Factory mFactory;
+	private List<File> mInputSources;
+	private String[] mInputClasspath;
+	private Map<File, File> mCorrespondanceMap;
+
+	private File getCorrespondingClassFile(File inputFile) {
+		String name = FilenameUtils.getBaseName(inputFile.getName());
+		Collection<File> classFiles = FileUtils.listFiles(mOldBinaryOutputDirectory, new NameFileFilter(name + ".class"), TrueFileFilter.INSTANCE);
+
+		if (classFiles.isEmpty()) {
+			return null;
+		}
+
+		for (File classFile : classFiles) {
+			String relative = mOldBinaryOutputDirectory.toPath().toAbsolutePath().relativize(classFile.toPath().toAbsolutePath()).toString();
+			String suffix = FilenameUtils.removeExtension(relative) + ".java";
```

Comment:
How this trick works when you have a compilation unit with multiples classes? Or an inner class?

Reply:
That's a good point, thanks. I'll try to make a better algorithm for matching .java files and .class files.
Also, if some file A.java was not changed, we need to add not only A.class into classpath, but all A$...class as well.

Reply:
You can also use: https://github.com/INRIA/spoon/blob/master/src/main/java/spoon/reflect/cu/CompilationUnit.java#L63. I added this function exactly for this purpose.

Reply:
@msteinbeck, Thank you, that's very handy.